mypyc/transform/uninit.py

```diff
@@ -45,7 +46,8 @@ def split_blocks_at_uninits(env: Environment,
                 # initialized is an operand to something other than a
                 # check that it is defined, insert a check.
                 if (isinstance(src, Register) and src not in defined
-                        and not (isinstance(op, Branch) and op.op == Branch.IS_ERROR)):
+                        and not (isinstance(op, Branch) and op.op == Branch.IS_ERROR)
+                        and not isinstance(op, LoadAddress)):
```

Comment:
Why do we need this? Add comment explaining this, since this is a bit surprising. I wonder if we can somehow get rid of this special case.

Reply:
We need this because we are taking the address of an uninitialized object, the automatically inserted initialization check will produce incorrect code, leading the control flow directly to uninitialized error rather than the following load address and other ops.